src/main/java/com/puppycrawl/tools/checkstyle/MultiThreadModeSettings.java

```diff
@@ -0,0 +1,89 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * A settings for the multi thread mode.
+ * @author Andrew Kuchev
+ */
+public class MultiThreadModeSettings implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /** The checker threads number. */
+    private final int checkerThreadsNumber;
+    /** The tree walker threads number. */
+    private final int treeWalkerThreadsNumber;
+
+    /**
+     * Initializes the multi thread mode configuration.
+     * @param checkerThreadsNumber the Checker threads number
+     * @param treeWalkerThreadsNumber the TreeWalker threads number
+     */
+    public MultiThreadModeSettings(int checkerThreadsNumber, int treeWalkerThreadsNumber) {
+        this.checkerThreadsNumber = checkerThreadsNumber;
+        this.treeWalkerThreadsNumber = treeWalkerThreadsNumber;
+    }
+
+    /**
+     * Gets the number of threads for the Checker module.
+     * @return the number of threads for the Checker module.
+     */
+    public int getCheckerThreadsNumber() {
+        return checkerThreadsNumber;
+    }
+
+    /**
+     * Gets the number of threads for the TreeWalker module.
+     * @return the number of threads for the TreeWalker module.
+     */
+    public int getTreeWalkerThreadsNumber() {
+        return treeWalkerThreadsNumber;
+    }
+
+    /**
+     * Creates a settings object which will enable single thread mode for
+     * Checker and TreeWalker modules.
+     * @return created settings
+     */
+    public static MultiThreadModeSettings getSingleThreadMode() {
+        return new MultiThreadModeSettings(1, 1);
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
A test is using PowerMock to mock a constructor (https://github.com/checkstyle/checkstyle/pull/4420/files#diff-06fde9f319ac6e31f8a808d413fe1f8eR593) and it is using equals to compare arguments